File: Regression-Models/Simple-Linear-Regression/simple_linear_regression.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling ---> Not required for this machine
# training_set = scale(training_set
# test_set = scale(test_set)

# fitting Simple Linear Regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)
# Predicting the Test set results
y_pred = predict(regressor, newdata = test_set)
# visualizing the Training set results
# install.packages('ggplot2') --> if required

library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = 'red') + 
  geom_line (aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
  color = 'blue') + 
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

# visualizing the test set results

library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = 'green') + 
  geom_line (aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
             color = 'blue') + 
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

File: Regression-Models/Simple-Linear-Regression/simple_linear_regression.py
# Simple Linear Regression

# Step 1 - Preprocessing

# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Step 2 - Fitting Simple Linear Regression to the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

# Step 3 - Predicting the Test Set Results
y_pred = regressor.predict(X_test)

# Step 4 - Visualizing the Training Set results
#plots
plt.scatter(X_train, y_train, color = 'red')
plt.scatter(X_train, regressor.predict(X_train), color = 'blue')
# title / labels of chart
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()
# Step 5 - Visualizing the Test Set results
#plots
plt.scatter(X_test, y_test, color = 'green')
plt.scatter(X_train, regressor.predict(X_train), color = 'blue') # don't change -> already trained
# title / labels of chart
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()